File: domain/peca.py
from dataclasses import dataclass
from datetime import datetime
from decimal import Decimal
from uuid import UUID


@dataclass
class Peca:
    id: UUID
    nome: str
    # Texto livre (ex: "Honda CG 160", "2020-2024") — futuro campo de busca
    # do RAG mencionado no CLAUDE.md.
    marca_modelo_compativel: str
    ano_compativel: str
    preco: Decimal  # nunca float — dinheiro não pode ter erro de arredondamento
    quantidade_estoque: int
    criado_em: datetime
    imagem_url: str | None = None
    quantidade_minima: int = 0  # 0 = sem alerta de estoque baixo configurado
    # None = peça não tem variação de cor (ex: vela, cabo) — nem toda peça
    # tem, por isso opcional, não um valor tipo "sem cor" fixo.
    cor: str | None = None

File: frontend/src/types/peca.ts
export interface Peca {
  id: string;
  nome: string;
  marca_modelo_compativel: string;
  ano_compativel: string;
  preco: string;
  quantidade_estoque: number;
  quantidade_minima: number;
  imagem_url: string | null;
  // Nem toda peça tem variação de cor (ex: vela, cabo) — por isso opcional.
  cor: string | null;
  criado_em: string;
}

export interface PecaCreateInput {
  nome: string;
  marca_modelo_compativel: string;
  ano_compativel: string;
  preco: string;
  quantidade_estoque: number;
  imagem_url?: string | null;
  cor?: string | null;
}
